src/lib/features/project/project-service.ts

```diff
@@ -324,17 +324,19 @@ export default class ProjectService {
             this.isEnterprise &&
             this.flagResolver.isEnabled('createProjectWithEnvironmentConfig')
         ) {
-            // todo: this is a workaround for backwards compatibility
-            // (i.e. not breaking enterprise tests) that we can change
-            // once these changes have been merged and enterprise
-            // updated. Instead, we can exit early if there are no cr
-            // envs
-            const crEnvs = newProject.changeRequestEnvironments || [];
-            await this.validateEnvironmentsExist(crEnvs.map((env) => env.name));
-            const changeRequestEnvironments =
-                await enableChangeRequestsForSpecifiedEnvironments(crEnvs);
-
-            data.changeRequestEnvironments = changeRequestEnvironments ?? [];
+            if (newProject.changeRequestEnvironments) {
+                await this.validateEnvironmentsExist(
+                    newProject.changeRequestEnvironments.map((env) => env.name),
+                );
+                const changeRequestEnvironments =
+                    await enableChangeRequestsForSpecifiedEnvironments(
+                        newProject.changeRequestEnvironments,
+                    );
+
+                data.changeRequestEnvironments = changeRequestEnvironments;
+            } else {
+                data.changeRequestEnvironments = [];
+            }
```

Comment:
As mentioned in the description, this could be this instead (essentially the same as before, but without the comment and without extra set of `??`s. Happy to hear your thoughts. 

```suggestion
            const crEnvs = newProject.changeRequestEnvironments ?? []
            await this.validateEnvironmentsExist(crEnvs.map((env) => env.name));
            const changeRequestEnvironments =
                await enableChangeRequestsForSpecifiedEnvironments(crEnvs,);

            data.changeRequestEnvironments = changeRequestEnvironments;
```

I kinda like this version, because it avoids all the nesting and it reads cleaner to me 🤷🏼

Reply:
More verbose version looks more readable